ygot/render.go

```diff
@@ -651,8 +668,8 @@ func EncodeTypedValue(val interface{}, enc gnmipb.Encoding) (*gnmipb.TypedValue,
 	switch {
 	case util.IsValueNil(vv) || !vv.IsValid():
 		return nil, nil
-	case vv.Type().Kind() == reflect.Int64:
-		// Invalid int64 that is not an enum.
+	case vv.Type().Kind() == reflect.Int64 && unionTypedefUnderlyingTypes[vv.Type().Name()] == nil:
+		// Invalid int64 that is not an enum or a union typedef type.
```

Comment:
`or a simple union typedef type (i.e., a union subtype that does not have a struct wrapper)`?